softvis3d-frontend/src/classes/TreeElement.ts

```diff
@@ -0,0 +1,56 @@
+export class TreeElement {
+
+    public readonly id: string;
+    public readonly key: string;
+    public measures: MeasureList;
+
+    public name: string;
+    public readonly path: string;
+
+    public parent?: TreeElement;
+    public readonly children: TreeElement[];
+
+    public isFile: boolean;
+
+    public constructor(id: string, key: string, measures: MeasureList, name: string, path: string, isFile: boolean,
+                       parent?: TreeElement) {
+        this.id = id;
+        this.key = key;
+        this.measures = measures;
+        this.name = name;
+        this.path = path;
+        this.isFile = isFile;
+        this.parent = parent;
+        this.children = [];
+    }
+
+    /**
+     * Will be called with the path of the components sorted.
+     */
+    public addAsChild(element: TreeElement, isDescending: boolean = false) {
+        let children = this.children;
+        if (isDescending) {
+            children = children.reverse();
+        }
+
+        for (let child of children) {
```

Comment:
`if (this.path === element.path) {`

Does never work (at least with the SQ data) as every element in the tree has a unique path and every element is only added once. this.path === element.path would never be true. The question here would be "is it inside the current node or should it be deeper in the tree" which could be something like:

`let remainingPath: string = element.path.substr(this.path.length + 1, element.path.length);
        if (remainingPath.indexOf("/") !== 0) {...`

I tried it with the above code but it did not work as expected. I would like to leave it as is.